src/cargo/util/command_prelude.rs

```diff
@@ -1027,6 +1036,37 @@ pub fn lockfile_path(
     return Ok(Some(path));
 }
 
+fn get_examples_candidates() -> Vec<clap_complete::CompletionCandidate> {
+    match get_examples_from_cwd() {
+        Ok(examples) => examples
+            .into_iter()
+            .map(|example| clap_complete::CompletionCandidate::new(example))
+            .collect(),
+        Err(_) => vec![],
+    }
+}
+
+fn get_examples_from_cwd() -> CargoResult<Vec<OsString>> {
```

Comment:
We need this to be reworked like how I suggested we re-work bins, which will make this PR build on top of that.